trieste/objectives/multi_objectives.py

```diff
@@ -38,7 +39,21 @@ def __call__(self, n: int, seed: int | None = None) -> TensorType:
         :param n: The number of pareto optimal points to be generated.
         :param seed: An integer used to create a random seed for distributions that
          used to generate pareto optimal points.
-        :return: The Pareto optimal points
+        :return: The Pareto optimal points.
+        """
+
+
+class Constraint(Protocol):
```

Comment:
@uri-granta we are introducing new interfaces here, it would be good if you could check them out?
they seem fine to me, but you'll know better

the only thing I can think of is that constraint functions could be in their own file as they could be used in other types of problems, not just multi objectives

Reply:
Fine to leave it here for now I think, as we can easily move it "outwards" later if we want to.

Another question is the name: is this purporting to be a general interface for constraints everywhere, or is it specific to specifying test problems?